tests/v2/validation/provisioning/rke2/post_kdm_oob_release_test.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type KdmChecksTestSuite struct {
+	suite.Suite
+	session                *session.Session
+	client                 *rancher.Client
+	ns                     string
+	rke2kubernetesVersions []string
+	cnis                   []string
+	providers              []string
+	nodesAndRoles          []machinepools.NodeRoles
+}
+
+func (k *KdmChecksTestSuite) TearDownSuite() {
+	k.session.Cleanup()
+}
+
+func (k *KdmChecksTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	k.session = testSession
+
+	k.ns = defaultNamespace
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	k.rke2kubernetesVersions = clustersConfig.RKE2KubernetesVersions
+
+	k.cnis = clustersConfig.CNIs
+	k.providers = clustersConfig.Providers
+	k.nodesAndRoles = clustersConfig.NodesAndRoles
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(k.T(), err)
+
+	k.client = client
+}
+
+func (k *KdmChecksTestSuite) TestRKE2K8sVersions() {
+	logrus.Infof("checking for valid k8s versions..")
+	require.GreaterOrEqual(k.T(), len(k.rke2kubernetesVersions), 1)
+	// fetching all available k8s versions from rancher
+	releasedK8sVersions, _ := versions.ListRKE2AllVersions(k.client)
+	logrus.Info("expected k8s versions : ", k.rke2kubernetesVersions)
+	logrus.Info("k8s versions available on rancher server : ", releasedK8sVersions)
+	for _, expectedK8sVersion := range k.rke2kubernetesVersions {
+		require.Contains(k.T(), releasedK8sVersions, expectedK8sVersion)
+	}
+}
+
+func (k *KdmChecksTestSuite) TestProvisioningSingleNodeRKE2Clusters() {
+	require.GreaterOrEqual(k.T(), len(k.providers), 1)
+	require.GreaterOrEqual(k.T(), len(k.cnis), 1)
+
+	subSession := k.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := k.client.WithSession(subSession)
+	require.NoError(k.T(), err)
+
+	kubeProvisioningClient, err := k.client.GetKubeAPIProvisioningClient()
+	require.NoError(k.T(), err)
+
+	providerName := k.providers[0]
+	provider := CreateProvider(providerName)
+	nodeRoles := k.nodesAndRoles
+
+	clusterNames := []string{}
+	clusterResps := []*v1.SteveAPIObject{}
+	k8sVersions := []string{}
+
+	// provisioning rke2 clusters using all released k8s versions
+	for _, k8sVersion := range k.rke2kubernetesVersions {
+
+		clusterName := namegen.AppendRandomString(provider.Name)
+
+		logrus.Info("provisioning " + k8sVersion + " cluster..")
```

Comment:
Can you move this logging after the call to `CreateK3SRKE2Cluster` function.

Reply:
done